druid-shell/src/platform/x11/clipboard.rs

```diff
@@ -231,6 +351,245 @@ impl Clipboard {
+                    return Ok(());
+                }
+                Some((atom, _, data)) => {
+                    if data.len() > maximum_property_length(conn) {
+                        // We need to do an INCR transfer.
+                        debug!("Starting new INCR transfer");
+                        let transfer = IncrementalTransfer::new(
+                            conn,
+                            event,
+                            Rc::clone(&data),
+                            self.atoms.INCR,
+                        );
+                        match transfer {
+                            Ok(transfer) => self.incremental.push(transfer),
+                            Err(err) => {
+                                reject_transfer(conn, event)?;
+                                return Err(err.into());
+                            }
+                        }
+                    } else {
+                        // We can provide the data directly
+                        conn.change_property8(
+                            PropMode::REPLACE,
+                            event.requestor,
+                            event.property,
+                            *atom,
+                            data,
+                        )?;
+                    }
+                }
+            }
+        }
+
+        // Inform the requestor that we sent the data
+        debug!("Replying to selection request event");
+        let event = SelectionNotifyEvent {
+            response_type: SELECTION_NOTIFY_EVENT,
+            sequence: 0,
+            requestor: event.requestor,
+            selection: event.selection,
+            target: event.target,
+            property: event.property,
+            time: event.time,
+        };
+        conn.send_event(false, event.requestor, EventMask::NO_EVENT, &event)?;
+
+        Ok(())
+    }
+
+    fn handle_property_notify(&mut self, event: PropertyNotifyEvent) -> Result<(), ReplyOrIdError> {
+        fn matches(transfer: &IncrementalTransfer, event: PropertyNotifyEvent) -> bool {
+            transfer.requestor == event.window && transfer.property == event.atom
+        }
+
+        if event.state != Property::DELETE {
+            return Ok(());
+        }
+        // Deleting the target property indicates that an INCR transfer should continue. Find that
+        // transfer
+        if let Some(transfer) = self
+            .incremental
+            .iter_mut()
+            .find(|transfer| matches(transfer, event))
+        {
+            let done = transfer.continue_incremental(&*self.connection)?;
+            if done {
+                debug!("INCR transfer finished");
+                // Remove the transfer
+                self.incremental
+                    .retain(|transfer| !matches(transfer, event));
+            }
+        }
+        Ok(())
+    }
+}
+
+#[derive(Debug)]
+struct ClipboardContents {
+    owner_window: Window,
+    data: Vec<(Atom, String, Rc<Vec<u8>>)>,
```

Comment:
Can we use `Rc<[u8]>`?

Reply:
I needed a moment to figure out how to construct that, but yeah, it seems to work.